### PR TITLE
make address information optional

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
makes the address information in state optional.  If they select "no" to "has my address changed", they'll be able to navigate throughout the application without being redirected back to the update address route upon reaching the review information screen.

### Related Azure Boards Work Items
[AB#4585](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/4585)

### Checklist

- [ ] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [ ] I have checked that my code follows the project's coding style by running `npm run format:check`
- [ ] I have checked that my code contains no linting errors by running `npm run lint`
- [ ] I have checked that my code contains no type errors by running `npm run typecheck`
- [ ] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/route-helpers/renew-route-helpers.server.ts [hunk 1]
```diff
@@ -31,17 +31,17 @@ export interface RenewState {
   };
   readonly hasAddressChanged?: boolean;
   readonly addressInformation?: {
-    copyMailingAddress: boolean;
+    copyMailingAddress?: boolean;
```
Comment (sebastien-comeau) [inline frontend/app/route-helpers/renew-route-helpers.server.ts:34]: Don't set the required address information props to `optional`.

File: frontend/app/.server/domain/entities/benefit-renewal.entity.ts [hunk 2]
```diff
@@ -20,26 +20,26 @@ export type BenefitRenewalRequestEntity = Readonly<{
       }>;
       PersonContactInformation: ReadonlyArray<
         Readonly<{
-          Address: ReadonlyArray<
+          Address?: ReadonlyArray<
```
Comment (nicholas-ly) [inline frontend/app/.server/domain/entities/benefit-renewal.entity.ts:23]: Should this Entity have an optional `Address` (same with the fields that were changed below and the DTO)? 

When a renewal application is submitted, the request to the downstream API should always have an address set. If the user didn't fill out the address information form, the application would grab the address on file for the client and send that instead.

File: frontend/app/routes/public/renew/$id/ita/confirmation.tsx
Comment (sebastien-comeau) [inline frontend/app/routes/public/renew/$id/ita/confirmation.tsx:1]: Addresses should only be resolved if `hasAddressChanged` is `true`. Same for review page.

**Review by sebastien-comeau — APPROVED** [review]
In another PR, make sure to only display or set the addresses if `hasAddressChanged` is `true`.

**Pull request merged**